file-validator.py

```diff
@@ -0,0 +1,239 @@
+from datetime import date
+from pydantic import BaseModel
+from typing import List
+import yaml
+import json
+import os
+from azure.storage.blob import BlobServiceClient
+from PIL import Image
+
+def fill_index_system(filename):
+     if is_json_empty(filename):
+        with open(filename, 'w') as json_file:
+            json.dump(default_json_data, json_file, indent=2)
+
+class ClientData(BaseModel):
+    clientEmail: str
+    clientExpertise: str
+
+class SeedData(BaseModel):
+    seedID: int
+    seedFamily: str
+    seedGenus: str
+    seedSpecies: str
+
+class ImageDataindex(BaseModel):
+    numberOfImages: int
+    
+class AuditTrail(BaseModel):
+    uploadDate: date
+    editedBy: str
+    editDate: date
+    changeLog: str
+    accessLog: str
+    privacyFlag: bool 
+
+class Index(BaseModel):
+    clientData: ClientData
+    imageData: ImageDataindex
+    seedData: SeedData
+    
+class PIndex(BaseModel):
+    clientData: ClientData
+    imageData: ImageDataindex
+    seedData: SeedData
+    auditTrail: AuditTrail
+    
+class UserData(BaseModel):
+    description: str
+    numberOfSeeds: int
+    zoom: float    
+    
+class Picture(BaseModel):
+    userData: UserData 
+    
+class ClientFeedback(BaseModel):
+    correctIdentification: bool
+    historicalComparison: str
+    
+    
+def is_yaml_file(file_path):
+    _, file_extension = os.path.splitext(file_path)
+    return file_extension.lower() == '.yaml'
+
+def manualMetaDataImport():
+    #ask name for the folder
+    folderName = input("Folder Name: ")
+    os.makedirs(folderName)
+    #build index
+    nbPic=buildIndex(folderName)
+    #for each picture in field
+    zoomlevel = input("Zoom level for this index: ")
+    seedNumber = input("Number of seed for this index: ")
+    for x in range(nbPic):
+        buildPicture(folderName,zoomlevel,seedNumber)
+
+def buildIndex(output:str):
+    # Ask for the url in the console
+
+    # # Retrieve some client info from the URL
+
+    # Ask Data (I dont have access to the data url atm)
+    clientEmail=input("clientEmail: ")
+    clientExpertise= input("Expertise: ")
+    clientData=ClientData(clientEmail=clientEmail,clientExpertise=clientExpertise)
```

Comment:
you're inconsistent in how you write your code. here the whitespaces before adn after the = signs (the rule is spaces except for parameters to fucntions). ruff extension should be able to give you feedback here.